spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/StackdriverLoggingAutoConfiguration.java

```diff
@@ -42,6 +43,7 @@
 @ConditionalOnClass({HandlerInterceptorAdapter.class, LoggingAppender.class})
 @ConditionalOnMissingBean(type = "org.springframework.cloud.sleuth.autoconfig.SleuthProperties")
 @AutoConfigureAfter(StackdriverTraceAutoConfiguration.class)
+@ConditionalOnWebApplication
```

Comment:
The problem that this might not be just enough. 
See its JavaDocs:
```
 * {@link Conditional} that matches when the application is a web application. By default,
 * any web application will match but it can be narrowed using the {@link #type()}
 * attribute.
```

I don't think that `TraceIdLoggingWebMvcInterceptor` is what we expect in the non-`SERVLET` environment...